actors/builtin/multisig/multisig_actor.go

```diff
@@ -89,13 +95,17 @@ func (a *MultiSigActor) Propose(rt vmr.Runtime, params *ProposeParams) *ProposeR
 	txnID := st.NextTxnID
 	st.NextTxnID += 1
 
-	st.PendingTxns = setPendingTxn(rt, st.PendingTxns, txnID, MultiSigTransaction{
+	actorHamt := adt.NewHamt(vmr.AsStore(rt), st.PendingTxns)
+	if err := actorHamt.Put(txnID, &MultiSigTransaction{
 		To:       params.To,
 		Value:    params.Value,
 		Method:   params.Method,
 		Params:   params.Params,
 		Approved: []addr.Address{callerAddr},
-	})
+	}); err != nil {
+		rt.AbortStateMsg(err.Error())
```

Comment:
Please don't use AbortStateMsg, I'm trying to kill it. Use Abort().

We might need a new error code in common.go, ErrIllegalState

Reply:
I've added the `ErrIllegalState` to common.go and instumented it for hamt operation failures in this file.